src/MoBi.Core/Services/SimulationLoader.cs

```diff
@@ -68,14 +72,26 @@ private void addSimulationToProject(IMoBiSimulation simulation, MoBiMacroCommand
 
          addSimulationConfigurationToProject(moBiSimulation, loadCommand);
 
-         moBiSimulation.ResultsDataRepository = simulation.ResultsDataRepository;
-         if (!_nameCorrector.CorrectName(project.Simulations, moBiSimulation))
-            return;
-
          moBiSimulation.HasChanged = true;
          loadCommand.AddCommand(new AddSimulationCommand(moBiSimulation));
       }
 
+      private void renameCollidingModules(IEnumerable<Module> modulesToRename, string simulationName, IReadOnlyList<IWithName> existingEntities)
+      {
+         var takenNames = existingEntities.AllNames();
+
+         modulesToRename
+            .Where(x => takenNames.Contains(x.Name))
+            .Select((x, index) => new { Module = x, Index = index })
+            .Each(item => item.Module.Name = getModuleNameByIndex(simulationName,item.Index) );
```

Comment:
What will happen if a simulation is loaded and there is a colliding simulation name, but no colliding module names? I think the loaded modules will not be renamed right?

In this step here, you want to check that any modules which were named after the old simulation name are renamed to the new simulation name even if there isn't already an existing module in the project with that name already. I would do that with a string replace because the modules being loaded might already be indexed.

Then you are correctly indexing if a collision is introduced on the renamed modules (below)